### ISSUE TITLE
widget-bound selections with spaces in their names lead to an error

### ISSUE THREAD

**Issue by jakevdp** [vega/vega-lite#5560]
Minimal spec demonstrating this:
```python
{
  "data": {"url": "data/cars.json"},
  "mark": "text",
  "selection": {
    "my selection": {
      "type": "single",
      "bind": {"input": "range"}
    }
  }
}
```
```
[Error] Cannot read property 'on' of undefined
```
If you change ``my selection`` to ``my_selection`` the error goes away. First reported in https://github.com/altair-viz/altair/issues/1791

**Comment by domoritz** [comment]
Currently selections have to be valid variable names. 

I wonder whether we should remove this limitation or provide another way to give a human readable name to a selection. What do you think @arvind?

**Comment by jakevdp** [comment]
FWIW, this does not cause an error in previous vega-lite releases (e.g. the set of deps used by the current altair release: vega-embed=5, vega=4, vega-lite=3.4)

**Comment by domoritz** [comment]
Then we should treat this as a regression.

**Comment by arvind** [comment]
Ugh, sorry about that. We did have unit tests for this issue (which was fixed in 29988ad502d896030b4c325f0b38742d4747b44e) but not for this specific code path which introduced a change when we disabled direct manipulation. Fixed with #5561.

**Comment by jakevdp** [comment]
Thanks!
